src/components/Swap/SwapActionButton/index.tsx

```diff
@@ -35,8 +37,32 @@ export default function SwapActionButton() {
   const srcWalletConnected = isStarknetChain(inputCurrency?.chainId) ? snAccount : account
   const dstWalletConnected = isStarknetChain(outputCurrency?.chainId) ? snAccount : account
 
+  const { toTokens, fromTokens } = useAtomValue(widgetSettingsAtom)
+  const fromTokensCommonChainId =
+    fromTokens &&
+    fromTokens[0] &&
+    (fromTokens.every((t) => t.chainId === fromTokens[0].chainId) ? fromTokens[0].chainId : undefined)
+  const toTokensCommonChainId =
+    toTokens &&
+    toTokens[0] &&
+    (toTokens.every((t) => t.chainId === toTokens[0].chainId) ? toTokens[0].chainId : undefined)
+  const isToTokensEVM = toTokens && toTokens.every((t) => !isStarknetChain(t.chainId))
+  const isFromTokensEVM = fromTokens && fromTokens.every((t) => !isStarknetChain(t.chainId))
+
   if (!inputCurrency && !outputCurrency) {
-    return <SwapButton disabled={isDisabled} />
+    if (fromTokensCommonChainId && (isStarknetChain(fromTokensCommonChainId) ? !snAccount : !account)) {
+      // If the 'From' tokens are on the same chain then a 'Connect' button with this chain should appear
+      return <ConnectWalletButton chainId={fromTokensCommonChainId} />
+    } else if ((isToTokensEVM || isFromTokensEVM) && !account) {
+      // If the 'From' or 'To' tokens are on the EVM chain then a 'Connect' button with the Ethereum chain should appear
+      return <ConnectWalletButton chainId={1} />
```

Comment:
if we have a 'From' token selected, shouldn't the chain here be the "fromTokenChainId" ?

Reply:
All these cases apply when the user does not select any tokens. At this point, we do not know which chain the user will choose, but we still want to present the connect button to them.

Reply:
oh ok! then :rocket: